### PR TITLE
[tests-only] Add spaces as a new WebDAV url type

### LINKED ISSUE #2854: add spaces WebDAV URL as a third variant for WebDAV path
Part of #2808

currently we use `old` and the `new` WebDAV path in the tests
old: `remote.php/webdav`
new: `remote.php/dav/files/<username>`

now we also need to run the tests with the new spaces concept.
The "private" space should react the same as the other WebDAV urls `/dav/spaces/<uuid>/`

The Example tables of the webdav related tests need to get an extra line. e.g.
```
  Scenario Outline: upload a file and check download content
    Given using <dav_version> DAV path
    When user "Alice" uploads file with content "uploaded content" to <file_name> using the WebDAV API
    Then the content of file <file_name> for user "Alice" should be "uploaded content"
    Examples:
      | dav_version | file_name      |
      | old         | "C++ file.cpp" |
      | old         | "file #2.txt"  |
      | new         | "C++ file.cpp" |
      | new         | "file #2.txt"  |

    @skipOnOcV10
    Examples:
      | spaces      | "C++ file.cpp" |
      | spaces      | "file #2.txt"  |

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added support for spaces WebDAV type in e2e test scenarios
- uses `acceptance-test-changes-waiting-2021-11` as base branch

## Related Issue
- https://github.com/owncloud/ocis/issues/2854

## How Has This Been Tested?

- :robot: 
- https://github.com/owncloud/ocis/pull/2887

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiComments-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34044/51/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavPreviews-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34048/95/1

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by SwikritiT — COMMENTED** [review]
lgtm

**Comment by phil-davis** [comment]
Converted to draft. I will review and cherry-pick into #39514 this morning.

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -2528,6 +2528,31 @@ public function getStatusPhp():ResponseInterface {
...
+		}
+		$response = HttpRequestHelper::get(
+			$fullUrl,
+			$this->getStepLineRef(),
+			$user,
+			$password
+		);
+		$json = \json_decode($response->getBody()->getContents());
+		// assuming the first space is the personal space
+		return $json->value[0]->id;
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/FeatureContext.php:2553]: In the next iteration we should double-check that the space is "personal". For now, these tests are only running in an environment that has a personal space.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**